rust-workspace/crates/ntc-asset-manager/src/algorand.rs

```diff
@@ -0,0 +1,126 @@
+use crate::Secret;
+
+/// Aliased re-export.
+pub use algonaut::error::AlgonautError;
+/// Aliased re-export.
+pub use algonaut::{
+    core::Address as AlgorandAddress,
+    model::algod::v2::TransactionResponse as AlgorandTransactionResponse,
+    transaction::{
+        error::TransactionError as AlgorandTransactionError, CreateAsset as AlgorandCreateAsset,
+        SignedTransaction as AlgorandSignedTransaction, Transaction as AlgorandTransaction,
+    },
+};
+/// Aliased re-export.
+pub use algonaut_client::error::ClientError as AlgonautClientError;
+
+// Import *real* struct when building
+#[cfg(not(test))]
+use algonaut::transaction::account::Account as AlgorandAccount;
+
+// Import *mock* struct when testing
+#[cfg(test)]
+use tests::MockAlgorandAccount as AlgorandAccount;
+
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum AlgorandSigningError {
+    #[error("conversion error: {0:?}")]
+    ConversionError(#[from] std::convert::Infallible),
+    #[error(transparent)]
+    TransactionError(#[from] AlgorandTransactionError),
+}
+
+type AlgorandSecret = Secret<32>;
+pub fn sign_transaction(
+    secret: AlgorandSecret,
+    txn: AlgorandTransaction,
+) -> Result<AlgorandSignedTransaction, AlgorandSigningError> {
+    let account = match secret {
+        AlgorandSecret::Mnemonic(val) => AlgorandAccount::from_mnemonic(&val)?,
+        AlgorandSecret::Seed(val) => AlgorandAccount::from_seed(*val),
+    };
+    let signed_txn = account.sign_transaction(&txn)?;
+
+    Ok(signed_txn)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use algonaut::transaction::{transaction::Payment, Transaction, TxnBuilder};
```

Comment:
Can we move this import into the `use algonaut` block below?

Reply:
Not sure that is possible, since `algonaut` is an external crate.  That is granted I understand your comment correctly.